core/lib/refinery/crud.rb

```diff
@@ -83,6 +83,13 @@ def create
           def edit
             # object gets found by find_#{singular_name} function
           end
+          
+          def duplicate
+            original = @#{singular_name}
+            @#{singular_name} = original.dup
```

Comment:
seeing as `original` is a variable that is never re-used this could be something like 
```ruby
def duplicate
  @#{singular_name} = @#{singular_name}.dup
  @#{singular_name}.title << ' Copy'
  render 'edit'
end
```